core/domain/beam_job_services.py

```diff
@@ -37,17 +38,33 @@
 datastore_services = models.Registry.import_datastore_services()
 
 
-def run_beam_job(job_name: str) -> beam_job_domain.BeamJobRun:
+def run_beam_job(
+    job_name: Optional[str] = None,
+    job_class: Optional[Type[base_jobs.JobBase]] = None,
+    run_synchronously: Optional[bool] = None
+) -> beam_job_domain.BeamJobRun:
     """Starts a new Apache Beam job and returns metadata about its execution.
 
     Args:
-        job_name: str. The name of the job to run.
+        job_name: str. The name of the job to run. If not provided, then
+            job_class must not be None.
+        job_class: type(JobBase). A subclass of JobBase to begin running. This
+            value takes precedence over job_name.
+        run_synchronously: bool. Whether to run the job synchronously. If not
+            provided, then the value is decided by whether Oppia is running in
+            emulator mode.
 
     Returns:
         BeamJobRun. Metadata about the run's execution.
     """
-    job_class = jobs_registry.get_job_class_by_name(job_name)
-    run_synchronously = constants.EMULATOR_MODE
+    if job_class is None and job_name is None:
+        raise ValueError('Must specify the job class to run')
+    if job_class is None:
+        # MyPy is wrong. We know job_name is not None in this branch because if
+        # it were, the ValueError above would have been raised.
+        job_class = jobs_registry.get_job_class_by_name(job_name) # type: ignore[arg-type]
+    if run_synchronously is None:
+        run_synchronously = constants.EMULATOR_MODE
```

Comment:
Do we want to provide this option? I don't see the usage rn.

Reply:
Sorry not sure what you mean? This option is used to decide between production behavior and dev behavior (production = EMULATOR_MODE = false = async job; dev = EMULATOR_MODE = true = sync job)

Reply:
I mean if we want to provide the option to override this by specifying `run_synchronously`, I think we should always follow `EMULATOR_MODE`.

Reply:
Done.